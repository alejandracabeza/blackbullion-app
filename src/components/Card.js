import React, { useState } from "react";
import Modal from "./Modal";

function Card({ data: dataCards }) {
  let [cardModal, setCardModal] = useState({});
  let [isShowModal, setIsShowModal] = useState(false);

  function handleClick(val) {
    console.log("s");
    setCardModal(val);
    setIsShowModal(!isShowModal);
  }
  return (
    <div>
      {dataCards &&
        dataCards.map((eachCard, index) => {
          return (
            <div className="eachCard" key={index}>
              <div className="imageContainer">
                <img src={eachCard.image} alt="information course" />
                <div className="duration">
                  <p>{eachCard.duration} min</p>
                </div>
              </div>
              <div className="cardInfo">
                <h2>{eachCard.title}</h2>
                <p>{eachCard.description}</p>
              </div>
              <div className="seeMore" onClick={() => handleClick(eachCard)}>
                See more...
              </div>
              <div className="startPathway">
                <a href={eachCard.url} target="_blank" rel="noreferrer">
                  Start Pathway <i className="fa fa-arrow-right"></i>
                </a>
              </div>
            </div>
          );
        })}
      {isShowModal && <Modal data={cardModal} onClick={handleClick} />}
    </div>
  );
}

export default Card;
