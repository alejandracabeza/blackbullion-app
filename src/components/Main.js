import React, { useState, useEffect } from "react";
import Card from "./Card";
import SearchBar from "./SearchBar";
import { getData } from "../api/index";
import SortButton from "./SortButton";

function Main() {
  let [data, setData] = useState([]);
  const [input, setInput] = useState("");
  const [dataFiltered, setDataFiltered] = useState([]);
  const [isSorted, setSortButton] = useState(false);

  useEffect(() => {
    getData().then((data) => {
      return setDataFiltered(data), setData(data);
    });
  }, []);

  async function updateInput(input) {
    if (data) {
      const filtered = data.filter((i) => {
        return (
          i.title.toLowerCase().includes(input) ||
          i.description.toLowerCase().includes(input)
        );
      });
      setInput(input);
      setDataFiltered(filtered);
    }
  }

  async function handleSortButton() {
    setSortButton(!isSorted);
    let sortedData = dataFiltered.sort(function (a, b) {
      return isSorted ? b.duration - a.duration : a.duration - b.duration;
    });
    setDataFiltered([...sortedData]);
  }

  return (
    <div className="main">
      {data.length ? (
        <div>
          <SearchBar input={input} onChange={updateInput} />
          <SortButton onClick={handleSortButton} />
          {dataFiltered.length &&
            dataFiltered.map((card) => {
              return <Card data={card} />;
            })}
        </div>
      ) : (
        <div className="loading"></div>
      )}
    </div>
  );
}

export default Main;
