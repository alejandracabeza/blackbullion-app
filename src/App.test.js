import React from "react";
import { shallow, mount } from "enzyme";
import Main from "./components/Main";
import Card from "./components/Card";

const dataCards = [
  {
    title: "Peer Pressure and Advertising",
    duration: "130",
    description:
      "Advertising is something that we are all surrounded by - online, watching TV or driving past a billboard. It can influence us and those around us more than we might realise. In this video, Joe and Jade consider some of the issues around advertising but also peer-pressure before they go shopping. They realise how important it to start assessing what you want to take control of their money.  ",
    url: "https://www.blackbullion.com/pathways/key-stage-4",
    image: "https://prodcontent.blackbullion.com/images/featured/lesson2.png",
  },
  {
    title: "Needs, Wants and Budgeting",
    duration: "269",
    description:
      "There are many things that we often consider as ‘needs’ but it might be worth taking a moment to think about whether this is what they actually are. You might find after a little thinking that it’s just a ‘want’. Jade and Joe discuss the financial implications of this in their finance vlog and the importance of budgeting. ",
    url: "https://www.blackbullion.com/pathways/key-stage-4",
    image: "https://prodcontent.blackbullion.com/images/featured/lesson3.png",
  },
];

describe("rendering components", () => {
  it("renders Main component without crashing", () => {
    shallow(<Main />);
  });
  it("renders Card component without crashing", () => {
    shallow(<Card />);
  });
});

describe("passing props", () => {
  const cardsWrappper = mount(<Card data={dataCards} />);
  it("accepts user account props", () => {
    expect(cardsWrappper.props().data).toEqual(dataCards);
  });
});
