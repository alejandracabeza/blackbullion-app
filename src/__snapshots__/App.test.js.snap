// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Cards snapshots 1`] = `
<div>
  <div
    className="eachCard"
    key="0"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="information course"
        src="https://prodcontent.blackbullion.com/images/featured/lesson2.png"
      />
      <div
        className="duration"
      >
        <p>
          130
           min
        </p>
      </div>
    </div>
    <div
      className="cardInfo"
    >
      <h2>
        Peer Pressure and Advertising
      </h2>
      <p>
        Advertising is something that we are all surrounded by - online, watching TV or driving past a billboard. It can influence us and those around us more than we might realise. In this video, Joe and Jade consider some of the issues around advertising but also peer-pressure before they go shopping. They realise how important it to start assessing what you want to take control of their money.  
      </p>
    </div>
    <div
      className="seeMore"
      onClick={[Function]}
    >
      See more...
    </div>
    <div
      className="startPathway"
    >
      <a
        href="https://www.blackbullion.com/pathways/key-stage-4"
        rel="noreferrer"
        target="_blank"
      >
        Start Pathway 
        <i
          className="fa fa-arrow-right"
        />
      </a>
    </div>
  </div>
  <div
    className="eachCard"
    key="1"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="information course"
        src="https://prodcontent.blackbullion.com/images/featured/lesson3.png"
      />
      <div
        className="duration"
      >
        <p>
          269
           min
        </p>
      </div>
    </div>
    <div
      className="cardInfo"
    >
      <h2>
        Needs, Wants and Budgeting
      </h2>
      <p>
        There are many things that we often consider as ‘needs’ but it might be worth taking a moment to think about whether this is what they actually are. You might find after a little thinking that it’s just a ‘want’. Jade and Joe discuss the financial implications of this in their finance vlog and the importance of budgeting. 
      </p>
    </div>
    <div
      className="seeMore"
      onClick={[Function]}
    >
      See more...
    </div>
    <div
      className="startPathway"
    >
      <a
        href="https://www.blackbullion.com/pathways/key-stage-4"
        rel="noreferrer"
        target="_blank"
      >
        Start Pathway 
        <i
          className="fa fa-arrow-right"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`snapshots Main snapshot 1`] = `
<div
  className="main"
>
  <div
    className="loading"
  />
</div>
`;

exports[`snapshots SearchBar snapshot 1`] = `
<div
  className="searchBarContainer"
>
  <input
    className="searchBar"
    onChange={[Function]}
    placeholder="Search..."
  />
</div>
`;
